### DISCUSSION TITLE
Change grpc's max message size

### DISCUSSION THREAD

**Discussion by DrBamba** [milvus-io/milvus#16840] [Q&A and General discussion]
I am getting the following problem:
Failed to register node
ResourceExhausted desc = trying to send message larger than max (2655193 vs. 2097152)
However I can't seem to find anywhere in the configs a limitation to 2MB
It's always 100MB as the default config file
Using milvus 2.0.2

EDIT:
I read that the etcd default request size is 1.5 + 0.5 of grpc overhead
So I tried adding the flag `--max-request-bytes 4MB' but it didn't change the error 
It still claims the limit is 2MB

The error arrives from the datatcoord and causes it to be unavailable and then the proxy is not healthy.
Rest of the logs seem irrelevant

**Comment by xiaofan-luan** [comment]
2097152 seems to be the etcd size limit.
It seems that the etcd request size is too large, any detailed error log?

**Reply by DrBamba** [reply]
Can you direct as to where can I find the relevant configuration?
Also, isn't the etcd request size limit 1.5MB? In this case it is 2MB and I **didn't** specify for example the flag `--max-request-bytes 2MB`

**Reply by DrBamba** [reply]
I read that the etcd default request size is 1.5 + 0.5 of grpc overhead
So I tried adding the flag `--max-request-bytes 4MB' but it didn't change the error 
It still claims the limit is 2MB

**Reply by kgmiller** [reply]
@DrBamba - did you ever find a solution to this? I'm down in production and every mention of this issue seems to lead nowhere...

**Reply by DrBamba** [reply]
@kgmiller unfortunately I don't believe I found a solution. I ended up trying multiple versions of etcd, giving the etcd more resources, and eventually I reset all of the data, and as much as I can remember it did not happen again.
I did however, not too long after, run into #16804 at which point I decided to go back to using milvus 1 until milvus 2.1 comes out, hopefully in the following month, and hopefully more stable, as I understand there will be a change in the communication between services.
